great_expectations/data_context/data_context/cloud_data_context.py

```diff
@@ -338,49 +336,6 @@ def root_directory(self) -> Optional[str]:
         """
         return self._context_root_directory
 
-    def _instantiate_datasource_from_config_and_update_project_config(
```

Comment:
This method is no longer needed since creating a new datasource config is handled by the `DatasourceStore.set()` method in all cases.